### DISCUSSION TITLE
[ALG101] LIOJ 1008 有關 parseInt() 的問題

### DISCUSSION THREAD

**Discussion by PT-HUANG** [Lidemy/forum#300] [其他問題]
老師好，各位同學好
想請問在使用 parseInt時
依照MDN的語法說明: parseInt(string, radix);
前面傳入一個字串，後面可以指定進位方式
不曉得為為什麼在local端測試時
console.log(binary)會回傳NaN
(用Number(lines[0]).toString(2)倒是AC了...XD)

範例程式碼如下:
function solve(lines){
  let binary = parseInt(lines[0], 2)
  let result = 0
  for (let i=0; i<binary.length; i++){
    if(Number(binary[i]) === 1){
      result ++
    }
  }
  console.log(result)
}

**Comment by aszx87410** [comment]
`parseInt` 的意思是你要讀進一個字串，而字串的原始進位是什麼。所以用法是 `parseInt('101011', 2)` 這樣子，你想要把數字轉成 2 進位的話，用 `toString(2)` 是正確的

**Reply by PT-HUANG** [reply]
謝謝老師，豁然開朗
